menu.rb

```diff
@@ -0,0 +1,163 @@
+# menu.rb
+require_relative 'library_manager'
+require_relative 'input_validator'
+
+class Menu
+  def initialize(library_manager)
+    @library_manager = library_manager
+  end
+
+  def display_menu
+    puts "\nLibrary Management System"
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a teacher'
+    puts '4. Create a student'
+    puts '5. Create a book'
+    puts '6. Create a rental'
+    puts '7. List rentals for a person'
+    puts '8. Quit'
+    print 'Enter your choice: '
+  end
+
+  MENU_OPTIONS = {
+    1 => :list_all_books,
+    2 => :list_all_people,
+    3 => :create_teacher,
+    4 => :create_student,
+    5 => :create_book,
+    6 => :create_rental,
+    7 => :list_rentals_for_person,
+    8 => :quit
+  }.freeze
+
+  def handle_choice(choice)
+    if MENU_OPTIONS.key?(choice)
+      send(MENU_OPTIONS[choice])
+    else
+      puts 'Invalid choice. Please choose a valid option.'
+    end
+  end
+
+  def valid_choice?(choice)
+    (1..8).include?(choice)
+  end
+
+  def list_all_books
+    puts 'List of All Books:'
+    @library_manager.books.each do |book|
+      puts "#{book.title} by #{book.author}"
+    end
+  end
+
+  def list_all_people
+    puts 'List of All People:'
+    @library_manager.people.each do |person|
+      puts "#{person.name} (ID: #{person.id})"
+    end
+  end
+
+  def create_teacher
+    puts "Enter teacher's name:"
+    name = gets.chomp
+    puts "Enter teacher's age:"
+    age = gets.chomp
+    puts "Enter teacher's specialization:"
+    specialization = gets.chomp
+
+    if InputValidator.validate_integer_input(age)
+      teacher = @library_manager.create_person(age.to_i, name, specialization)
+      puts "Teacher #{teacher.name} (ID: #{teacher.id}) created successfully."
+    else
+      puts 'Invalid age. Please enter a valid age.'
+    end
+  end
+
+  def create_student
+    puts 'Enter student\'s name:'
+    name = gets.chomp
+    puts 'Enter student\'s age:'
+    age = gets.chomp
+
+    if InputValidator.validate_integer_input(age)
+      student = @library_manager.create_person(age.to_i, name)
+      puts "Student #{student.name} (ID: #{student.id}) created successfully."
+    else
+      puts 'Invalid age. Please enter a valid age.'
+    end
```

Comment:
- [ ] From the video prompt provided, you're meant to first create a person, then ask if the user is going to be a student or teacher, and ask for parent permission when creating a person. 
- In your case, you have separate options for the requirement, and your input does not ask for parent permission. Kindly fix this issue 👍 

### Video prompt:
![image](https://github.com/zbahati/school_library_oop/assets/50990925/59e1f2ae-6a41-4578-b6fb-c562a39742f3)

### Your view:
![image](https://github.com/zbahati/school_library_oop/assets/50990925/bf060d7a-0812-469f-aae3-c10fdf1f6594)